src/OpenSSL/crypto.py

```diff
@@ -1548,11 +1555,44 @@ def _init(self):
         """
         Set up the store context for a subsequent verification operation.
         """
+        self._cleanup()
+
+        # build OpenSSL STACK_OF(X509 *) structure containing the
+        # untrusted chain certs
+        chain_stack = _ffi.NULL
+        if self._chain:
+            chain_stack = _lib.sk_X509_new_null()
+            _openssl_assert(chain_stack != _ffi.NULL)
+
+            def cleanup(s):
+                # Equivalent to sk_X509_pop_free, but we don't
+                # currently have a CFFI binding for that available
+                for i in range(_lib.sk_X509_num(s)):
+                    x = _lib.sk_X509_value(s, i)
+                    _lib.X509_free(x)
+                _lib.sk_X509_free(s)
+            chain_stack = _ffi.gc(chain_stack, cleanup)
+
+            for cert in self._chain:
+                copy = _lib.X509_dup(cert._x509)
```

Comment:
I couldn't find a binding for X509_up_ref() in cryptography, and didn't want to make this diff depend on adding one. If this has changed, I'd be happy to make this change.